### PR TITLE
Update plugin lib.rs template

### PR DESCRIPTION
**Pull request by vsumner**
**Description:**
This PR updates the plugin template to match the changes from https://github.com/swc-project/swc/pull/4206

**BREAKING CHANGE:**

**Related issue (if exists):**

### CODE REVIEW

File: crates/swc_cli/src/commands/plugin.rs [hunk 1]
```diff
@@ -243,7 +243,7 @@ impl VisitMut for TransformVisitor {
 /// important steps manually need to be performed like sending transformed
 /// results back to host. Refer swc_plugin_macro how does it work internally.
 #[plugin_transform]
-pub fn process_transform(program: Program, _plugin_config: String, _context: String) -> Program {
+pub fn process_transform(program: Program, _metadata: TransformPluginProgramMetadata, _context: String) -> Program {
```
Comment (kwonoj) [inline crates/swc_cli/src/commands/plugin.rs:246]: `_context: String` also need to be removed - now all metadata is included in `TransformPluginProgramMetadata`.

**Review by kwonoj — COMMENTED** [review]
Thanks for catching this!

File: crates/swc_cli/src/commands/plugin.rs (hunk 1 above)
Comment (vsumner) [reply crates/swc_cli/src/commands/plugin.rs]: Thanks! I forgot to port that from my project.

**Review by kwonoj — APPROVED** [review]

**Review by kdy1 — APPROVED** [review]
Thanks!

---

swc-bump:
 - swc_cli

**Pull request merged**